hack/generator/pkg/astmodel/type_name.go

```diff
@@ -19,27 +19,27 @@ type TypeName struct {
 }
 
 // NewTypeName is a factory method for creating a TypeName
-func NewTypeName(pr PackageReference, name string) *TypeName {
-	return &TypeName{pr, name}
+func NewTypeName(pr PackageReference, name string) TypeName {
```

Comment:
minor, in a few other places rather than using `NewX` and returning ptr I saw somebody (Bevan?) use `MakeX` -- might be worth making that distinction here too?